Create Assignment Statuses

mzsanford opened the pull request:
Add the notion of status to the Assignment model This includes the data model change as well as the management UI changes. The inline forms were submitting via Turbo, which is so fast as to make it seems like they're not even being saved. I added a little "Saved" label to make it more clear.

Source: [Add all Assignment/Finisher statuses to the web app](https://looseendsproject.slack.com/lists/T05GP1TRWQM/F08B7PGPUMU?record_id=Rec08AZS1LVRD)

## Screen Shots

Project Show page

![CleanShot 2025-02-12 at 12 26 02@2x](https://github.com/user-attachments/assets/e575dbf1-cc4e-478e-b666-a827c07ea040)

Assignments Saving


https://github.com/user-attachments/assets/9b147cdc-bf4a-428e-ab71-cb758829d64e




Assignment Status at assigning time

![CleanShot 2025-02-11 at 11 46 23@2x](https://github.com/user-attachments/assets/3b5b0cd5-8df8-4aad-9e13-bd01526dbe0a)

robert-r-meyer reviewed (approved):
Changes look good to me. The tests are super nice to have. Thanks for the work.

swards commented:
@mzsanford - looks great.  Wondering why is there a save and delete button?  Could we do without them?

mzsanford commented:
> Wondering why is there a save and delete button? Could we do without them?

You know, @swards, that's a great question. I had the "delete" button to replicate the previous trash can. Your question got me thinking about it and the new status values of `declined` and `unresponsive` communicate better than that so it's not really needed. Removed.

Once delete was gone I asked the UX question of _why_ the user should care this is an underlying object. They shouldn't care so I changed to a more Apple style UX where changing the drop-down triggers the save. This along with the new "Saved" label work much better. I've added a video to the PR description showing it.

swards commented:
> > Wondering why is there a save and delete button? Could we do without them?
> 
> Removed.
> I changed to a more Apple style UX where changing the drop-down triggers the save.

Looks great!

robert-r-meyer commented:
@mzsanford got a merge conflict on the PR. Let me know if you need help sorting it out.

mzsanford commented:
I was out of town for the school break last week. I'll fix the conflicts and merge this tomorrow.
